### DISCUSSION TITLE
Looking for a way to update i18n messages

### DISCUSSION THREAD

**Discussion by sandrooco** [codex-team/editor.js#2876] [Q&A]
We have an internationalized application. Users can change languages with a simple drawer and the UI including the data will update without page reloads.

Angular translations and editor.js internationalization obviously isn't the same and editor.js instances won't be re-initialized. So they still contain the messages from the previous language. Because of that, I'm looking for a way to change the messages of an already running editor.js instance.
We tried to save the current editor's state and re-initialize editor.js but couldn't get it working reliably. This is why it would be great to have something like `editorInstance.setMessages(...)`.

I couldn't find anything similar in the github issues, maybe also just failed to search properly. Is this complex to implement or any other workarounds available?

**Comment by neSpecc** [comment]
Why not just re-init editorjs? Updating messages will require re-rendering of the UI

**Comment by sandrooco** [comment]
Thanks for your quick feedback!
Because there might be changes in the editor already, so we would have to first extract the data and re-initialize (which we tried but found no working and quick solution).
Setting the messages and re-rendering sounds quite feasible to me.
